### PR TITLE
Editor: allow pasting in image captions

### PR DESCRIPTION
**Pull request by michaeldcain**
Fixes: #10254

This updates the wp-editimage TinyMCE plugin to reflect the changes in: https://core.trac.wordpress.org/changeset/38756

captions

Props @azaozz 

**To test:**
1. Check out the branch locally or use calypso.live
2. Create a new post or edit an existing post with an image.
3. Add a caption to the image by clicking the caption icon.
4. Select the placeholder caption text and paste in a new caption from your clipboard.

### CODE REVIEW

**Comment by azaozz** [comment]
Looks good. Tested pasting different content copied from different places, all works as expected (block tags are removed and <br> inserted instead, inline tags are kept).

**Comment by azaozz** [comment]
Still looks good after killing all `var` and adding "mutant" constant variables (lol that sounds almost as weird as a `constant` that is not "constant", but coding standards and lint settings have to be respected).

Had some troubles with undo after pasting into multiple image captions. Couldn't consistently reproduce.

**Comment by marekhrabe** [comment]
The code looks good and the functionality too. I've seen inconsistent undo bugs all over the editor and I think that is something to be tackled on a higher level.

File: client/components/tinymce/plugins/wpeditimage/plugin.js [hunk 1]
```diff
@@ -305,13 +311,13 @@ function wpEditImage( editor ) {
 		} );
 
 		editor.on( 'wpImageFormSubmit', function( event ) {
-			var data = event.imgData.data,
+			const data = event.imgData.data;
+			let	wrap, parent, node, html, imgId,
```
Comment (aduth) [inline client/components/tinymce/plugins/wpeditimage/plugin.js]: Tab after `let` should be a space.

**Comment by aduth** [comment]
>but coding standards and lint settings have to be respected).

In at least one case [we've just outright disabled linting](https://github.com/Automattic/wp-calypso/blob/11cf35d9521346214016b7d3f7d546853e7af30d/client/components/tinymce/plugins/wpcom/plugin.js#L20). Guess it comes down to how often we expect to need to keep these in sync, vs. willingness to wholly support a fork or go through a manual reconciliation process. But generally it'd be best to not have to disable the linter.

**Comment by aduth** [comment]
Also fixes: #1658
Previous effort: #3986

**Pull request merged**